src/test/debugger/android/androidPlatform.test.ts

```diff
@@ -0,0 +1,280 @@
+                    message: message.message,
+                    args: message.args.filter(value => value),
+                };
+            });
+            messagesWithoutUndefineds.should.eql([expectedMessage]);
+        }
+
+        function shouldHaveReceivedNoLogCatMessages(): void {
+            fakeExtensionMessageSender.getAllMessagesSent().should.eql([]);
+        }
+
+        const testWithRecordings = new RecordingsHelper(() => reactNative).test;
+
+        testWithRecordings("runApp launches the app when a single emulator is connected",
+            [
+                "react-native/run-android/win10-rn0.21.0/succeedsWithOneVSEmulator",
+                "react-native/run-android/win10-rn0.22.2/succeedsWithOneVSEmulator",
+                "react-native/run-android/osx10.10-rn0.21.0/succeedsWithOneVSEmulator",
+            ], () => {
+                return Q({})
+                    .then(() => {
+                        return simulatedAVDManager.createAndLaunch("Nexus_5");
+                    }).then(() => {
+                        return androidPlatform.runApp();
+                    }).then(() => {
+                        return deviceHelper.isAppRunning(androidPackageName);
+                    }).then(isRunning => {
+                        isRunning.should.be.true();
+                        shouldHaveReceivedSingleLogCatMessage("Nexus_5");
+                    });
+            });
+
+        testWithRecordings("runApp launches the app when two emulators are connected",
+            ["react-native/run-android/win10-rn0.21.0/succeedsWithTwoVSEmulators"], () => {
+                return Q({})
+                    .then(() => {
+                        return simulatedAVDManager.createAndLaunchAll("Nexus_5", "Nexus_6");
+                    }).then(() => {
+                        return androidPlatform.runApp();
+                    }).then(() => {
+                        return Q.all([
+                            deviceHelper.isAppRunning(androidPackageName, "Nexus_5"),
+                            deviceHelper.isAppRunning(androidPackageName, "Nexus_6"),
+                        ]);
+                    }).spread((isRunningOnNexus5, isRunningOnNexus6) => {
+                        // It should be running in exactly one of these two devices
+                        isRunningOnNexus5.should.not.eql(isRunningOnNexus6);
+                        const emulatorWithAppRunningId = isRunningOnNexus5 ? "Nexus_5" : "Nexus_6";
+                        shouldHaveReceivedSingleLogCatMessage(emulatorWithAppRunningId);
+                    });
+            });
+
+        testWithRecordings("runApp launches the app when three emulators are connected",
+            ["react-native/run-android/win10-rn0.21.0/succeedsWithThreeVSEmulators"], () => {
+                return Q({})
+                    .then(() => {
+                        return simulatedAVDManager.createAndLaunchAll("Nexus_5", "Nexus_6", "Other_Nexus_6");
```

Comment:
Maybe change `createAndLaunchAll()` to take an array instead? That way, you can store the array locally in this test, and avoid recreating the array further below in this test. Makes it easier to change the tested device IDs in my opinion